aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -333,7 +333,7 @@ static void and_kernel_impl(TensorIterator& iter) {
     binary_kernel_reduce_vec(
         iter,
         [=](uint8_t a, uint8_t b) -> uint8_t { return (a && b) ? 1 : 0; },
-#if defined(CPU_CAPABILITY_ZVECTOR)
+#if defined(CPU_CAPABILITY_AVX2) || defined(CPU_CAPABILITY_AVX512) || defined(CPU_CAPABILITY_ZVECTOR)
```

Comment:
Can we just remove the `#else` here?

Reply:
I think it's for VSX

Reply:
Correct, it's for VSX.

Reply:
I agree that VSX would hit that path, but should it not just fall back to the default implementation of `operator&` in that case?

Reply:
Tried to remove `#else` branches, but it ends up throwing compile errors, like:

```
git/aten/src/ATen/native/cpu/Reduce.h:47:17: error: ‘vop’ cannot be used as a function
   47 |     acc[0] = vop(acc[0], Vec::loadu(ptr + (0 * Vec::size() * sizeof(scalar_t))));
      |              ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Just so we're on the same page, the following code fails to compile?
```cpp
    binary_kernel_reduce_vec(
        iter,
        [=](uint8_t a, uint8_t b) -> uint8_t { return (a && b) ? 1 : 0; },
        [=](Vectorized<uint8_t> a, Vectorized<uint8_t> b) {
          return a & b;
        },
        /*ident=*/true);
```

Reply:
Sorry, my bad - just pushed the right change, it's running CI now.